src/topp/MultiplexResolver.cpp

```diff
@@ -0,0 +1,674 @@
+  But let us now assume the sequence "LDNLVAIFDINR(Label:13C(6)15N(4))" is mapped to the heavy partner of a peptide feature pair. This is no conflict. Medium and heavy
+  peptides have been correctly detected. The MultiplexResolver adds a dummy peptide feature of zero intensity at the light position and thereby completes the triplet.
+
+  <B>The command line parameters of this tool are:</B>
+  @verbinclude TOPP_MultiplexResolver.cli
+  <B>INI file documentation of this tool:</B>
+  @htmlinclude TOPP_MultiplexResolver.html
+
+*/
+
+// We do not want this class to show up in the docu:
+/// @cond TOPPCLASSES
+
+class TOPPMultiplexResolver :
+  public TOPPBase
+{
+private:
+
+  // input and output files
+  String in_;
+  String out_;
+  String out_conflicts_;
+
+  // section "algorithm"
+  String labels_;
+  std::vector<std::vector<String> > samples_labels_;
+  unsigned missed_cleavages_;
+  double mass_tolerance_;
+
+  // section "labels"
+  map<String, double> label_mass_shift_;
+  
+public:
+  TOPPMultiplexResolver() :
+    TOPPBase("MultiplexResolver", "Completes peptide multiplets and resolves conflicts within them.", true),
+    missed_cleavages_(0), mass_tolerance_(0.1)
+  {
+  }
+  
+  void registerOptionsAndFlags_()
+  {
+    registerInputFile_("in", "<file>", "", "Peptide multiplets with assigned sequence information");
+    setValidFormats_("in", ListUtils::create<String>("consensusXML"));
+    registerOutputFile_("out", "<file>", "", "Complete peptide multiplets.", false);
+    setValidFormats_("out", ListUtils::create<String>("consensusXML"));
+    registerOutputFile_("out_conflicts", "<file>", "", "Optional output containing peptide multiplets with conflicting quant/ID information.", false, true);
+    setValidFormats_("out_conflicts", ListUtils::create<String>("consensusXML"));
+
+    registerSubsection_("algorithm", "Parameters for the algorithm.");
+    registerSubsection_("labels", "Isotopic labels that can be specified in section \'algorithm:labels\'.");
+
+  }
+
+  // create parameters for sections (set default values and restrictions)
+  Param getSubsectionDefaults_(const String& section) const
+  {
+    Param defaults;
+
+    if (section == "algorithm")
+    {
+      defaults.setValue("labels", "[][Lys8,Arg10]", "Labels used for labelling the samples. [...] specifies the labels for a single sample. For example\n\n[][Lys8,Arg10]        ... SILAC\n[][Lys4,Arg6][Lys8,Arg10]        ... triple-SILAC\n[Dimethyl0][Dimethyl6]        ... Dimethyl\n[Dimethyl0][Dimethyl4][Dimethyl8]        ... triple Dimethyl\n[ICPL0][ICPL4][ICPL6][ICPL10]        ... ICPL");
+      defaults.setValue("missed_cleavages", 0, "Maximum number of missed cleavages due to incomplete digestion.");
+      defaults.setMinInt("missed_cleavages", 0);
+      defaults.setValue("mass_tolerance", 0.1, "Mass tolerance in Da for matching the detected to the theoretical mass shifts.", ListUtils::create<String>("advanced"));
+    }
+
+    if (section == "labels")
```

Comment:
this seems duplicated from the FeatureFinderMultiplex.
Please remove this duplication if possible.